addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java

```diff
@@ -181,7 +181,7 @@ protected Object convertToType(String value) {
         } else if (value.equalsIgnoreCase("false") || value.equalsIgnoreCase("off")) {
             return (Boolean.FALSE);
         } else if (value.matches("(-|\\+)?[0-9]+")) {
-            return (Integer.valueOf(value));
+            return Integer.valueOf((Long.valueOf(value).intValue()));
```

Comment:
Imho the following should be enough, although it still makes an overflow mess at runtime:

```java
Long.valueOf(value).intValue();
```